ngraph/test/type_prop/ctc_loss.cpp

```diff
@@ -0,0 +1,320 @@
+    auto ctc_loss =
+        make_shared<op::v4::CTCLoss>(logits, logit_length, labels, label_length, blank_index);
+
+    // check type and shape infer
+    EXPECT_EQ(ctc_loss->get_element_type(), element::f32);
+    EXPECT_TRUE(ctc_loss->get_output_partial_shape(0).same_scheme(PartialShape{10}));
+}
+
+TEST(type_prop, ctc_loss_fail_inputs_dim)
+{
+    // create inputs
+    auto logits = make_shared<op::Parameter>(element::f32, Shape{10, 120, 40, 28});
+    auto logit_length = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto labels = make_shared<op::Parameter>(element::i32, Shape{10, 120});
+    auto label_length = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto blank_index = make_shared<op::Parameter>(element::i32, Shape{});
+
+    try
+    {
+        // create CTCLoss node
+        auto ctc_loss =
+            make_shared<op::v4::CTCLoss>(logits, logit_length, labels, label_length, blank_index);
+
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Invalid inputs not detected";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("Expected a 3D tensor for logits."));
+    }
+    catch (...)
+    {
+        FAIL() << "Inputs shape check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, ctc_loss_fail_logit_length_dim)
+{
+    // create inputs
+    auto logits = make_shared<op::Parameter>(element::f32, Shape{10, 120, 28});
+    auto logit_length = make_shared<op::Parameter>(element::i32, Shape{10, 20});
+    auto labels = make_shared<op::Parameter>(element::i32, Shape{10, 120});
+    auto label_length = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto blank_index = make_shared<op::Parameter>(element::i32, Shape{});
+
+    try
+    {
+        // create CTCLoss node
+        auto ctc_loss =
+            make_shared<op::v4::CTCLoss>(logits, logit_length, labels, label_length, blank_index);
+
+        // Should have thrown, so fail if it didn't
+        FAIL() << "Invalid logit length not detected";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), std::string("Expected a 1D tensor for logit length."));
+    }
+    catch (...)
+    {
+        FAIL() << "Logit length shape check failed for unexpected reason";
+    }
+}
+
+TEST(type_prop, ctc_loss_fail_labels_dim)
+{
+    // create inputs
+    auto logits = make_shared<op::Parameter>(element::f32, Shape{10, 120, 28});
+    auto logit_length = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto labels = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto label_length = make_shared<op::Parameter>(element::i32, Shape{10});
+    auto blank_index = make_shared<op::Parameter>(element::i32, Shape{});
+
+    try
+    {
+        // create CTCLoss node
+        auto ctc_loss =
+            make_shared<op::v4::CTCLoss>(logits, logit_length, labels, label_length, blank_index);
```

Comment:
Please use EXPECT_THROW instead of try ... catch

Reply:
fixed

Reply:
Oh, ok. Do we really need to check the error message?

Reply:
I suppose it was made to check certain scenario: such inputs lead to appropriate message.